scalalib/src/mill/scalalib/TestModuleUtil.scala

```diff
@@ -355,16 +362,35 @@ private final class TestModuleUtil(
       }
     }
 
+    val filteredClassCount = filteredClassLists.map(_.size).sum
     val executor = Executors.newScheduledThreadPool(1)
     val outputs =
       try {
-        // Periodically check the queueLog file of every runner, and tick the executing test name
+        // Periodically check the claim log and claim stats files of every runner, and tick the relevant infos
         executor.scheduleWithFixedDelay(
-          () =>
+          () => {
+            // reuse to reduce syscall, may not be as accurate as running `System.currentTimeMillis()` inside each entry
+            val now = System.currentTimeMillis()
             workerStatusMap.forEach { (claimLog, callback) =>
               // the last one is always the latest
-              os.read.lines(claimLog).lastOption.foreach(callback)
-            },
+              os.read.lines(claimLog).lastOption.foreach { currentTestClass =>
+                testClassTimeMap.putIfAbsent(currentTestClass, now)
+                val last = testClassTimeMap.get(currentTestClass)
+                callback(s"$currentTestClass${mill.internal.Util.renderSecondsSuffix(now - last)}")
+              }
+            }
+            var totalSuccess = 0L
+            var totalFailure = 0L
+            workerResultSet.forEach { (claimStats, _) =>
+              val (success, failure) =
+                upickle.default.read[(Long, Long)](os.read.stream(claimStats))
+              totalSuccess += success
+              totalFailure += failure
+            }
+            ctx.log.ticker(
+              s"${totalSuccess + totalFailure}/${filteredClassCount} completed, ${totalFailure} failures"
```

Comment:
Let's only print `n failures` for `n > 0`, since that's what we do in the main prompt for failed tasks